### DISCUSSION TITLE
Validation on last step of Wizard not being called

### DISCUSSION THREAD

**Discussion by jgarvey-dev** [filamentphp/filament#13395] [Help] [answered]
### Package

Form builder

### Package Version

3.2

### How can we help you?

I have a wizard with 3 steps and the last step does not trigger any method for validation.

Both step 1 and step 2 will trigger the beforeValidation and afterValidation methods.

The last step only has the save button assigned with the submitAction method which does not trigger the beforeValidation and afterValidation methods.

I have also tried using:
protected function beforeSave(): void
protected function mutateFormDataBeforeSave(array $data): array
public function save(bool $shouldRedirect = true, bool $shouldSendSavedNotification = true): void

None of these are triggered when I hit the save button.

Is there any way to perform a final validation before saving?

**Comment by G1ET1N** [comment]
Try 

```
public function create() : void
{
//... your logic
}
```

**Comment by leandrocfe** [comment]
Are you using a custom livewire component or a resource?

**Reply by jgarvey-dev** [reply, marked as answer]
Turns out it was an issue in my view file

I had 
`<form wire:submit="update">`
and should have
`<form wire:submit="save">`

And not  having an update method in my class was causing Model->update() to be called which does not have any of the hooks provided.

**Reply by EyeOfProvidence7** [reply]
Hey man, the code blocks are missing for me and I need this fix lol, could you share them again if you have them?

**Reply by alisilarbi** [reply]
> Are you using a custom livewire component or a resource?

Hello, I have the same issue, and I'm usig a custom livewire component, I have 2 checkbox and I would like to validate them upon clicking on submit button. Upong clicking on it it doesnt validate the checkboxes. Any tip ?

**Reply by jgarvey-dev** [reply]
> Hey man, the code blocks are missing for me and I need this fix lol, could you share them again if you have them?

should see them now....
